### ISSUE TITLE
genreport resource does not validate parameters

### ISSUE THREAD

**Issue by smithfarm** [smithfarm/dochazka-rest#53]
The `genreport` resource (POST only) takes an entity with a number of parameters. The `path` parameter is mandatory - this is used to fetch the relevant `component` object.  The template objectified by the `component` object may take parameters. All parameters other than `path` are passed to the template. However, there is no way to validate the parameters - this makes it difficult or impossible to avoid errors like:

```
Dochazka(2016-01-09) root ADMIN> generate report sample/site_param.mc
*** REST ERROR ***

Error encountered on attempted operation "GENERATE REPORT"
REST operation: POST genreport
HTTP status: 500 Internal Server Error
Explanation: The 'explanation' parameter ("Attribute (param) is required at constructor MC2::sample_site_param_mc::new (defined at /usr/lib/perl5/vendor_perl/5.20.1/Mason/Interp.pm
line 309) line 40     MC2::sample_site_param_mc::new('MC2::sample_site_param_mc', 'm', 
'Moose::Meta::Class::__ANON__::SERIAL::5=HASH(0x5f828a0)') called at /usr/lib/perl5/vendor_perl/5.20.1/Mason/Request.pm line 222     
Mason::Request::construct_page_component('Moose::Meta::Class::__ANON__::SERIAL::5=HASH(0x5f828a0)', 
'MC2::sample_site_param_mc', 'HASH(0x5f84b18)') called at /usr/lib/perl5/vendor_perl/5.20.1
/Mason/Request.pm line 305     Mason::Request::__ANON__ at /usr/lib/perl5/vendor_perl/5.20.1
/Mason/Request.pm line 228    Mason::Request::try {...}  at /usr/lib/perl5/vendor_perl/5.20.1
/Try/Tiny.pm line 81     eval {...} at /usr/lib/perl5/vendor_perl/5.20.1/Try/Tiny.pm line 72        
Try::Tiny::try('CODE(0x5ed9d78)', 'Try::Tiny::Catch=REF(0x63f6f60)') called at /usr/lib/perl5
/vendor_perl/5.20.1/Mason/Request.pm line 238 
Mason::Request::catch_abort('Moose::Meta::Class::__ANON__::SERIAL::5=HASH(0x5f828a0)', 
'CODE(0x5f843c8)') called at /usr/lib/perl5/vendor_perl/5.20.1/Mason/Request.pm line 312       
Mason::Request::run('Moose::Meta::Class::__ANON__::SERIAL::5=HASH(0x5f828a0)', '/sample
/site_param') called at /usr/lib/perl5/vendor_perl/5.20.1/Mason/Interp.pm line 365  
Mason::Interp::run('Mason::Interp=HASH(0x5f82d50)', '/sample/site_param') called at /usr/lib/perl5
/site_perl/5.20.1/App/Dochazka/REST/Model/Component.pm line 360     
App::Dochazka::REST::Model::Component::generate('App::Dochazka::REST::Model::Component=H
ASH(0x5f67528)') called at /usr/lib/perl5/site_perl/5.20.1/App/Dochazka/REST/Dispatch.pm line 
1572 App::Dochazka::REST::Dispatch::handler_genreport('App::Dochazka::REST::Dispatch=HASH(0x5ca
ebc0)', 2) called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Entity.pm line 247   
Web::MREST::Entity::try {...}  at /usr/lib/perl5/vendor_perl/5.20.1/Try/Tiny.pm line 81    eval {...} at 
/usr/lib/perl5/vendor_perl/5.20.1/Try/Tiny.pm line 72     Try::Tiny::try('CODE(0x5f84098)', 
'Try::Tiny::Catch=REF(0x5f66d50)') called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Entity.pm 
line 265   
Web::MREST::Entity::mrest_generate_response_json('App::Dochazka::REST::Dispatch=HASH(0x5c
aebc0)') called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Entity.pm line 301   
Web::MREST::Entity::mrest_generate_response('App::Dochazka::REST::Dispatch=HASH(0x5caebc0)') called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Entity.pm line 200   
Web::MREST::Entity::mrest_process_request_json('App::Dochazka::REST::Dispatch=HASH(0x5caebc0)') called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Entity.pm line 221     
Web::MREST::Entity::mrest_process_request('App::Dochazka::REST::Dispatch=HASH(0x5caebc0)') called at /usr/lib/perl5/site_perl/5.20.1/Web/MREST/Resource.pm line 1036 W
```

(etc. etc.)

**Comment by smithfarm** [comment]
There are two problems here:

1. a future component might want to take a `path` parameter, which would conflict with the mandatory `path` property of the `genreport` resource. Solution: change it to `_path` or `+path`
2. the `component` object itself would need to include a list of parameters and their corresponding validations.

**Comment by smithfarm** [comment]
Fixed in 0.516:

```
Dochazka(2016-01-09) root ADMIN> generate report sample/site_param.mc
*** REST ERROR ***

Error encountered on attempted operation "GENERATE REPORT"
REST operation: POST genreport
HTTP status: 400 Bad Request
Explanation: Mandatory parameter 'param' missing in call to App::Dochazka::REST::Dispatch::handler_genreport
: Mandatory parameter 'param' missing in call to App::Dochazka::REST::Dispatch::handler_genreport
Permanent? YES
```
